react-app/src/components/projects/AddProject.tsx

```diff
@@ -0,0 +1,375 @@
+
+interface IDisposalProjectSearch {
+  rows: any[];
+  setRows: (a: any[]) => void;
+}
+
+type ParcelWithType = Parcel & {
+  Type: string;
+};
+
+type BuildingWithType = Building & {
+  Type: string;
+};
+
+const DisposalProjectSearch = (props: IDisposalProjectSearch) => {
+  const { rows, setRows } = props;
+  const [autoCompleteVal, setAutoCompleteVal] = useState(null);
+  const [fuzzySearchOptions, setFuzzySearchOptions] = useState([]);
+  const [loadingOptions, setLoadingOptions] = useState(false);
+  const api = usePimsApi();
+  const theme = useTheme();
+
+  const getAutoCompleteLabel = (input: ParcelWithType | BuildingWithType | string) => {
+    if (typeof input === 'string') {
+      return '';
+    }
+
+    if (input.PID) {
+      return `${input.Type} - PID: ${input.PID}`;
+    } else if (input.PIN) {
+      return `${input.Type} - PIN: ${input.PIN}`;
+    } else if (input.Address1) {
+      return `${input.Type} - Address: ${input.Address1}`;
+    } else {
+      return `${input.Type} - No identifying info.`;
+    }
+  };
+
+  const columns: GridColDef[] = [
+    {
+      field: 'Type',
+      headerName: 'Type',
+      flex: 1,
+    },
+    {
+      field: 'PID_Address',
+      headerName: 'PID/Address',
+      valueGetter: (params) => params.row.PID ?? params.row.Address1,
+      flex: 1,
+      renderCell: (params) => {
+        return (
+          <Link
+            to={`/properties/${params.row.Type.toLowerCase()}/${params.row.Id}`}
+            target="_blank"
+            rel="noopener noreferrer"
+            style={{ color: theme.palette.primary.main }}
+          >
+            {params.value}
+          </Link>
+        );
+      },
+    },
+    {
+      field: 'Agency',
+      headerName: 'Agency',
+      valueGetter: (params) => params.value?.Name ?? 'N/A',
+      flex: 1,
+    },
+    {
+      field: 'Year',
+      headerName: 'Year',
+      flex: 1,
+      valueGetter: (params) =>
+        (params.row.Evaluations as ParcelEvaluation[] | BuildingEvaluation[])
+          ?.map((a) => a.Year) //Map evaluations to just their year.
+          .sort((a, b) => b - a)?.[0] ?? 'N/A', //Sort in reverse order to obtain most recent year.
+    },
+    {
+      field: 'Evaluation',
+      headerName: 'Evaluation',
+      valueGetter: (params) => params.row.Evaluations?.[0]?.Value ?? 'N/A',
+      flex: 1,
+    },
+    {
+      field: 'Actions',
+      headerName: '',
+      flex: 1,
+      maxWidth: 60,
+      renderCell: (params) => {
+        return (
+          <IconButton
+            onClick={() => {
+              const index = rows.findIndex((x) => x.Id === params.row.Id);
+              if (index != null) {
+                setRows([...rows.slice(0, index), ...rows.slice(index + 1)]);
+              }
+            }}
+          >
+            <Delete />
+          </IconButton>
+        );
+      },
+    },
+  ];
+  return (
+    <Box display={'flex'} flexDirection={'column'} gap={'1rem'}>
+      <Autocomplete
```

Comment:
Suggest adding a freesolo prop. Otherwise I get console warnings about field contents not being options.